test/withGL.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+function withGL(init, fn)
+{
+    if (!PIXI.utils.isWebGLSupported())
+    {
+        return undefined;
+    }
+
+    const renderer = init();
+
+    try
+    {
+        return fn(renderer);
+    }
+    finally
+    {
+        PIXI.utils.destroyTextureCache();
+        renderer.destroy();
```

Comment:
That is one less tab-level, and that kind of abstractions are used widely in modern languages. Its like try-with-resources from java/scala.